wholesale/entity.py

```diff
@@ -0,0 +1,91 @@
+from common.constants import entity_key, cache_id
+from entity.abstract import ResourceEntity
+from datetime import date, datetime
+
+import re
+import json
+import requests
+
+API_URL = 'https://data.cityofnewyork.us/resource/87fx-28ei.json' + \
+    '?$select=`market`,`bic_number`,`account_name`,`application_type`,`postcode`' + \
+    '&$where=disposition_date >= \'{}\' and disposition_date <= \'{}\' &$limit=10000&$offset={}' + '&$order=disposition_date'
+
+ZIPCODE_PATTERN = re.compile('^\\d{5}$')
+ALPHA_ONLY_PATTERN = re.compile('^[A-Za-z\\s]+$')
+
+class WholesaleMarket(ResourceEntity):
+
+    @staticmethod
+    def dependencies():
+        return[
+            entity_key.census_us_zipcode,
+            entity_key.calendar_date,
+            entity_key.wholesale_market_app_type
+        ]
+
+    def get_zipcode_id(self, record, field):
+        zipcode_entity = self.dependencies_map[entity_key.census_us_zipcode]
+        zipcode = zipcode_entity.get_cached_value(record[field])
+        return zipcode['id'] if zipcode else None
+    
+    def get_market_app_type_id(self, record, field):
+        app_type_entity = self.dependencies_map[entity_key.wholesale_market_app_type]
+        app_type = app_type_entity.get_cached_value(record[field])
+        return app_type['id'] if app_type else None
+    
+    def to_lowercase(self, record, field):
+        return record[field].lower() if record[field] else None
+
+    def __init__(self):
+        super().__init__()
+
+        self.table_name = 'wholesale_market'
+        self.fields = [
+            {'field': 'market'},
+            {'field': 'bic_number', 'column': 'public_id'},
+            {'field': 'account_name', 'data': self.to_lowercase},
```

Comment:
We don't want to save account name as lower case to the database we only want to test against the lower case version of the value to determine uniqueness in case the casings vary between records. Also if we're still using `bic_number` as the public id then we don't need to do this at all.

We only want to alter the data if we need to alter it's type or map it to a foreign key. But string/ varchar data types should remain unchanged from the raw data.

Reply:
1. That makes sense
2. _"Also if we're still using bic_number as the public id then we don't need to do this at all."_  
Didn't we want to persist bic_number (public_id) in the record? That's what I have in my notes from the conversation we had. Should I remove it, or are you just mentioning it to support the account_name comment

Reply:
I think originally we were using it as a mark for uniqueness but the account name is probably more accurate.